components/brave_shields/browser/brave_shields_util.cc

```diff
@@ -332,25 +333,54 @@ void SetFingerprintingControlType(Profile* profile,
   if (!primary_pattern.IsValid())
     return;
 
-  auto* map = HostContentSettingsMapFactory::GetForProfile(profile);
-  map->SetContentSettingCustomScope(
-      primary_pattern, ContentSettingsPattern::Wildcard(),
-      ContentSettingsType::PLUGINS, kFingerprintingV2,
-      GetDefaultAllowFromControlType(type));
+  // Clear previous value to have only one rule for one pattern.
+  HostContentSettingsMapFactory::GetForProfile(profile)->
+      SetContentSettingCustomScope(
+          primary_pattern,
+          ContentSettingsPattern::FromString("https://balanced/*"),
+          ContentSettingsType::PLUGINS,
+          kFingerprintingV2,
+          CONTENT_SETTING_DEFAULT);
+  HostContentSettingsMapFactory::GetForProfile(profile)->
+      SetContentSettingCustomScope(primary_pattern,
+                                   ContentSettingsPattern::Wildcard(),
+                                   ContentSettingsType::PLUGINS,
+                                   kFingerprintingV2,
+                                   CONTENT_SETTING_DEFAULT);
+
+  auto content_setting = CONTENT_SETTING_BLOCK;
+  auto secondary_pattern =
+      ContentSettingsPattern::FromString("https://balanced/*");
+
+  if (type != ControlType::DEFAULT) {
+    content_setting = (type == ControlType::ALLOW) ? CONTENT_SETTING_ALLOW
```

Comment:
we have helpers for this `GetDefaultAllowFromControlType` and `GetDefaultBlockFromControlType`

Reply:
Done.